backend/tests/houseblockList.test.js

```diff
@@ -671,4 +673,337 @@ describe('Houseblock list controller', () => {
+        comment: 'comment13',
+        startTime: '2017-10-15T18:00:00',
+        endTime: '2017-10-15T18:30:00',
+        locationCode: 'UNEMPLOYED',
+      },
+      {
+        bookingId: 4,
+        offenderNo: 'A1234AD',
+        firstName: 'STEVE',
+        lastName: 'JONES',
+        cellLocation: 'LEI-A-1-4',
+        event: 'CHAP',
+        eventType: 'PRISON_ACT',
+        payRate: 1.3,
+        eventDescription: 'Chapel Activity',
+        comment: 'comment13',
+        startTime: '2017-10-15T18:00:00',
+        endTime: '2017-10-15T18:30:00',
+        locationCode: 'STAYONWING',
+      },
+      {
+        bookingId: 5,
+        offenderNo: 'A1234AE',
+        firstName: 'JOHN',
+        lastName: 'SMITH',
+        cellLocation: 'LEI-A-1-5',
+        event: 'CHAP',
+        eventType: 'PRISON_ACT',
+        payRate: 1.3,
+        eventDescription: 'Chapel Activity',
+        comment: 'comment13',
+        startTime: '2017-10-15T18:00:00',
+        endTime: '2017-10-15T18:30:00',
+        locationCode: 'RETIRED',
+      },
+    ]
+
+    it('should return only offenders leaving the wing', async () => {
+      elite2Api.getHouseblockList.mockImplementationOnce(() => responseWithOneLeavingWing)
+
+      const response = await houseblockList({}, 'LEI', 'Houseblock 1', '15/10/2017', 'ED', 'leaving')
+
+      expect(response.length).toEqual(1)
+      expect(response).toEqual([
+        {
+          activities: [
+            {
+              attendanceInfo: {
+                absentReason: undefined,
+                comments: undefined,
+                id: undefined,
+                locked: undefined,
+                paid: undefined,
+              },
+              bookingId: 2,
+              cellLocation: 'LEI-A-1-1',
+              comment: 'comment12',
+              endTime: '2017-10-15T18:30:00',
+              event: 'CHAP',
+              eventDescription: 'Chapel',
+              eventType: 'PRISON_ACT',
+              firstName: 'MICHAEL',
+              lastName: 'SMITH',
+              mainActivity: true,
+              offenderNo: 'A1234AB',
+              startTime: '2017-10-15T18:00:00',
+            },
+          ],
+          alertFlags: [],
+          bookingId: 2,
+          category: '',
+          cellLocation: 'LEI-A-1-1',
+          courtEvents: [],
+          eventId: undefined,
+          eventLocationId: undefined,
+          firstName: 'MICHAEL',
+          lastName: 'SMITH',
+          offenderNo: 'A1234AB',
+          releaseScheduled: false,
+          scheduledTransfers: [],
+          stayingOnWing: false,
+        },
+      ])
+    })
+
+    it('should return only offenders staying on the wing', async () => {
+      elite2Api.getHouseblockList.mockImplementationOnce(() => responseWithOneLeavingWing)
+
+      const response = await houseblockList({}, 'LEI', 'Houseblock 1', '15/10/2017', 'ED', 'staying')
+
+      expect(response.length).toEqual(4)
+      expect(response).toEqual([
```

Comment:
There's a lot of test data here, I wonder if it'll be easier to reason about if `expect.objectContaining` was used, to validate part of the data structure?  That way the dataset would be a bit clearer.